containenv/utils.py

```diff
@@ -0,0 +1,229 @@
+                         command.schema)
+            args.update(command.schema.validate(args))
+        sys.exit(command() or 0)
+    
+
+    def _get_command_module(self, command):
+        """Return the command module for the specified CLI command.
+    
+        Args:
+            command: The name of the CLI command to use in finding a module in the
+                commands package.
+    
+        Returns:
+            A module object for the given CLI command with a valid docopt __doc__
+            and a main() function that takes a list containing the command line
+            arguments.
+    
+        Raises:
+            ValueError: Raised if no module with a name matching the command
+                exists in the commands package.
+        """
+        mod_name = '{}.{}'.format(self.commands.__name__, command)
+        logging.debug('mod_name: {}'.format(mod_name))
+        try:
+            return importlib.import_module(mod_name)
+        except ImportError:
+            logger.exception('Unable to import "%s".', mod_name)
+            raise ValueError(
+                '"{}" is not a containenv command. \'containenv help -a\' lists '
+                'all available subcommands.'.format(command)
+            )
+    
+    
+    def _help(self, help_subject_cmd):
+        """Print out a help message and exit the program.
+    
+        Args:
+            help_subject_cmd: If a help_subject_cmd value is supplied then print
+                the help message for the command module if available. If the
+                help_subject_cmd is '-a' or '--all', then print the standard help
+                message but with a full list of available commands.
+    
+        Raises:
+            ValueError: Raised if the help message is requested for an invalid
+                command or an unrecognized option is passed to help.
+        """
+        logging.debug('Inside self._help command is {}'.format(help_subject_cmd))
+        if not help_subject_cmd:
+            print(self.module_doc.format(''))
+            sys.exit(0)
+        elif help_subject_cmd in ('-a', '--all'):
+            available_commands = (command_name for _, command_name, _ in
+                                  pkgutil.iter_modules(self.commands.__path__))
+            command_doc = 'Available commands:\n{}'.format(
+                '\n'.join(
+                    '   {}'.format(command) for command in available_commands)
+                )
+            print(self.module_doc.format(command_doc))
+            sys.exit(0)
+        elif help_subject_cmd.startswith('-'):
+            raise ValueError("Unrecognized option '{}'.".format(help_subject_cmd))
+        else:
+            mod = self._get_command_module(help_subject_cmd)
+            logging.debug("mod is {}".format(mod))
+            doc = mod.__doc__
+        logging.debug(doc)
+        docopt(doc, argv=('help',))
+            
+    
+    def _get_command(self, tokens):
```

Comment:
I still think the subcommand argument is the way to go, per the examples. This feels fragile.